fdbserver/workloads/GetMappedRange.actor.cpp

```diff
@@ -0,0 +1,405 @@
+#include "flow/Error.h"
+#include "flow/IRandom.h"
+#include "flow/flow.h"
+#include "ApiWorkload.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+const Value EMPTY = Tuple().pack();
+ValueRef SOMETHING = "SOMETHING"_sr;
+const KeyRef prefix = "prefix"_sr;
+const KeyRef RECORD = "RECORD"_sr;
+const KeyRef INDEX = "INDEX"_sr;
+
+struct GetMappedRangeWorkload : ApiWorkload {
+	bool enabled;
+	Snapshot snapshot = Snapshot::False;
+
+	//	const bool BAD_MAPPER = deterministicRandom()->random01() < 0.1;
+	const bool BAD_MAPPER = false;
+	//	const bool SPLIT_RECORDS = deterministicRandom()->random01() < 0.5;
+	const bool SPLIT_RECORDS = true;
+	const static int SPLIT_SIZE = 3;
+
+	GetMappedRangeWorkload(WorkloadContext const& wcx) : ApiWorkload(wcx) {
+		enabled = !clientId; // only do this on the "first" client
+	}
+
+	std::string description() const override { return "GetMappedRange"; }
+
+	Future<Void> start(Database const& cx) override {
+		// This workload is generated different from typical ApiWorkload. So don't use ApiWorkload::_start.
+		if (enabled) {
+			return GetMappedRangeWorkload::_start(cx, this);
+		}
+		return Void();
+	}
+
+	ACTOR Future<Void> performSetup(Database cx, GetMappedRangeWorkload* self) {
+		std::vector<TransactionType> types;
+		types.push_back(NATIVE);
+		types.push_back(READ_YOUR_WRITES);
+
+		wait(self->chooseTransactionFactory(cx, types));
+		return Void();
+	}
+
+	Future<Void> performSetup(Database const& cx) override { return performSetup(cx, this); }
+
+	Future<Void> performTest(Database const& cx, Standalone<VectorRef<KeyValueRef>> const& data) override {
+		// Ignore this because we are not using ApiWorkload's default ::start.
+		return Future<Void>();
+	}
+
+	static Key primaryKey(int i) { return Key(format("primary-key-of-record-%08d", i)); }
+	static Key indexKey(int i) { return Key(format("index-key-of-record-%08d", i)); }
+	static Value dataOfRecord(int i) { return Key(format("data-of-record-%08d", i)); }
+	static Value dataOfRecord(int i, int split) { return Key(format("data-of-record-%08d-split-%08d", i, split)); }
+
+	static Key indexEntryKey(int i) {
+		return Tuple().append(prefix).append(INDEX).append(indexKey(i)).append(primaryKey(i)).pack();
+	}
+	static Key recordKey(int i) { return Tuple().append(prefix).append(RECORD).append(primaryKey(i)).pack(); }
+	static Key recordKey(int i, int split) {
+		return Tuple().append(prefix).append(RECORD).append(primaryKey(i)).append(split).pack();
+	}
+	static Value recordValue(int i) { return Tuple().append(dataOfRecord(i)).pack(); }
+	static Value recordValue(int i, int split) { return Tuple().append(dataOfRecord(i, split)).pack(); }
+
+	ACTOR Future<Void> fillInRecords(Database cx, int n, GetMappedRangeWorkload* self) {
+		loop {
+			std::cout << "start fillInRecords n=" << n << std::endl;
+			// TODO: When n is large, split into multiple transactions.
+			state Transaction tr(cx);
+			try {
+				tr.reset();
```

Comment:
Delete this line, and move line 100 out of the loop. Basically `tr.onError` does the right thing for resetting the internal states of `tr` and updating the timeout value.

If calling `reset` here, the timeout value is also reset, which is undesirable.